### ISSUE TITLE
Protocol support for continuing previously started execution after client restart.

### ISSUE THREAD

**Issue by drykovanov** [jupyter/jupyter_client#746]
I was looking for a way to catch up previously started executions after notebook was closed and then re-opened.
As far as I understand currently it is not possible to continue receiving IOPub/execute_reply without issuing execute_request in current session.

This feature may be useful for cloud services in order to support long running executions.

Probably that could be achieved by introducing executions discovery mechanism and by allowing execute_request to have special flags requiring to attach to existing execution.

**Comment by davidbrochart** [comment]
Right, currently it doesn't seem to work. For instance, open the following notebook in Jupyter Lab in two different browsers:
```python
import time

i = 0
while True:
    print(i)
    i += 1
    time.sleep(1)
```
And run it from one browser. You don't see the counter being displayed in the other notebook.
It's the same issue as opening the notebook and running it, then closing it (but the kernel is still running) and reopening it. When you reopen it, it uses the same kernel, but you don't see the counter being displayed.
I'm not sure why, I think it should work. In any case, this will likely change with Real Time Collaboration, when we share the cell's outputs.
